tests/foreman/ui/test_puppetenvironment.py

```diff
@@ -38,20 +39,20 @@ def test_positive_end_to_end(session, module_org, module_location):
     """
     name = gen_string('alpha')
     new_name = gen_string('alpha')
-    with session:
+    with Session(hostname=session_puppet_enabled_sat.hostname) as session:
         session.puppetenvironment.create(
```

Comment:
Though it seems to work well, I somehow feel reluctant to rewrite all to this new pattern. Is it recommended or is there an effort to refactor UI tests to use the `ui_session` instead of `with Session`?

Reply:
@vsedmik I think you're right about the scope of this PR and the change over to using the session attribute. @JacobCallahan is this a change you would like to apply broadly across the UI tests?

Reply:
Ok folks you changed my mind. Pushed new version with `sat.ui_session`.
Tests are passing...
```
=== 9 passed, 62 warnings in 4763.49s (1:19:23) ===
```

Reply:
@vsedmik wish we had discussed it a bit more - you should still use a context with the session, the property isn't cached so every call to `sat.ui_session` is creating a new session instance and doing configuration.

It should look like:

```
with sat.ui_session as session:
    # same statements that the test had before using session.stuff
```

We're just making it so that you don't have to use the session fixture or any imports, and its an explicit resolution of a UI session instance configured for `default_sat`.

Reply:
@vsedmik shared offline that the property actually is cached! This actually causes tests after the first UI test to fail, because the context manager closes the browser attached to the session and never reopens it. 

This caching conflicts with robottelo's airgun integration and session handling, and I've asked him to remove that caching in this PR.